### PR TITLE
prometheus exporter: add units to metric names in TYPE and HELP comments

### PR DESCRIPTION
**Pull request by dashpole**
Fixes https://github.com/open-telemetry/opentelemetry-java/issues/5716

Changes:

* ensures the `_total` suffix is always added after unit suffixes by trimming it before adding unit suffixes.
* ensures the TYPE and HELP comments have the correct metric name, including unit suffixes.
* Update unit tests to use seconds, as ratio (1) is not common.

### CODE REVIEW

**Comment by dashpole** [comment]
Test failure seems unrelated: 

```
> Task :sdk:metrics:compileTestJava
/home/runner/work/opentelemetry-java/opentelemetry-java/sdk/metrics/src/test/java/io/opentelemetry/sdk/metrics/internal/descriptor/MetricDescriptorTest.java:86: error: cannot find symbol
                Advice.create(Arrays.asList(1.0, 2.0))));
                      ^
  symbol:   method create(List<Double>)
  location: class Advice
1 error

> Task :sdk:metrics:compileTestJava FAILED
```

File: exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/PrometheusMetricNameMapper.java [hunk 1]
```diff
@@ -42,14 +42,20 @@ private static String mapToPrometheusName(MetricData rawMetric, PrometheusType p
     String name = NameSanitizer.INSTANCE.apply(rawMetric.getName());
     String prometheusEquivalentUnit =
         PrometheusUnitsHelper.getEquivalentPrometheusUnit(rawMetric.getUnit());
+    boolean shouldAppendUnit =
+        !StringUtils.isNullOrEmpty(prometheusEquivalentUnit)
+            && !name.contains(prometheusEquivalentUnit);
+    // trim counter's _total suffix so the unit is placed before it.
+    if (prometheusType == PrometheusType.COUNTER && name.endsWith("_total")) {
+      name = name.substring(0, name.length() - 6);
```
Comment (mateuszrzeszutek) [inline exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/PrometheusMetricNameMapper.java:50]: Nit 
```suggestion
      name = name.substring(0, name.length() - "_total".length());
```

File: exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/Serializer.java [hunk 2]
```diff
@@ -568,7 +565,13 @@ String contentType() {
     }
 
     @Override
-    String headerName(String name, PrometheusType type) {
+    String headerName(String name, MetricData rawMetric, PrometheusType type) {
+      // If the name didn't originally have a _total suffix, and we added it later, omit it from the
+      // header.
+      String sanitizedOriginalName = NameSanitizer.INSTANCE.apply(rawMetric.getName());
+      if (!sanitizedOriginalName.endsWith("_total") && (type == PrometheusType.COUNTER)) {
+        return name.substring(0, name.length() - 6);
```
Comment (mateuszrzeszutek) [inline exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/Serializer.java:573]:
```suggestion
        return name.substring(0, name.length() - "_total".length());
```

**Review by mateuszrzeszutek — APPROVED** [review]

File: exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/PrometheusMetricNameMapper.java [hunk 3]
```diff
@@ -42,14 +42,20 @@ private static String mapToPrometheusName(MetricData rawMetric, PrometheusType p
     String name = NameSanitizer.INSTANCE.apply(rawMetric.getName());
     String prometheusEquivalentUnit =
         PrometheusUnitsHelper.getEquivalentPrometheusUnit(rawMetric.getUnit());
+    boolean shouldAppendUnit =
+        !StringUtils.isNullOrEmpty(prometheusEquivalentUnit)
+            && !name.contains(prometheusEquivalentUnit);
+    // trim counter's _total suffix so the unit is placed before it.
+    if (prometheusType == PrometheusType.COUNTER && name.endsWith("_total")) {
+      name = name.substring(0, name.length() - "_total".length());
```
Comment (jack-berg) [inline exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/PrometheusMetricNameMapper.java:50]: #nit: Let's extract "_total" to a private static final.

**Review by jack-berg — APPROVED** [review]

**Pull request merged**